src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -357,6 +357,8 @@ def setup_distributed(self):
 
     def setup(self, trainer: "pl.Trainer") -> None:
         self.accelerator.setup(trainer)
+        # we set the device so that optimizers can be created with distributed comms.
+        self.lightning_module._device = self.root_device
```

Comment:
does this only apply to deepspeed or could this be relevant to other strategies as well (e.g. sharded)? Just wondering if we should do this for all strategies

Reply:
Looking at other plugins, there isn't as much of a worry as they can move the model to device fine! I think its just the deepspeed plugin that has issues (maybe bagua? but seems the original PR took care)